### PR TITLE
Add priority servers, update servers_v7.json

### PR DESCRIPTION
**Pull request by BasedUser**
currently bound to a [martian](https://en.wikipedia.org/wiki/Martian_packet) ip address, will be using DNS control to mirror the discord events

If your pull request is **not** translation or serverlist-related, read the list of requirements below and check each box:

- [X] I have read the [contribution guidelines](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md).
- [X] I have ensured that my code compiles, if applicable.
- [X] I have ensured that any new features in this PR function correctly in-game, if applicable.

### CODE REVIEW

**Comment by Phinner** [comment]
Since this server will be used for events, it would be nice to add a priority field for it so it always comes on top of the list when the server is reachable 🤔.

**Comment by BasedUser** [comment]
> Since this server will be used for events, it would be nice to add a priority field for it so it always comes on top of the list when the server is reachable thinking.

hmmm m m

**Comment by BasedUser** [comment]
i haven't touched mindustry code in a very long time i'm gonna need some review help

**Comment by Ilya246** [comment]
PR compiled+tested ingame by me, priority servers work

**Comment by Phinner** [comment]
@BasedUser, I made a PR to your branch with some suggestions https://github.com/BasedUser/Mindustry/pull/6.

**Review by Phinner — APPROVED** [review]

File: core/src/mindustry/ui/dialogs/JoinDialog.java [hunk 1]
```diff
@@ -417,6 +417,9 @@ void refreshCommunity(){
 
             Table[] groupTable = {null, null};
 
+            if(group.prioritized){
+                addHeader(groupTable, group, hidden);
```
Comment (Anuken) [inline core/src/mindustry/ui/dialogs/JoinDialog.java:421]: I haven't tested this, but won't it add an empty, pointless group table if the event server is offline?

Comment (Phinner) [reply core/src/mindustry/ui/dialogs/JoinDialog.java:421]: OK, I tried to fix it, but now there are empty spaces where the unreachable prioritized servers are. Is there a way to make it truly invisible or I have to change the height manually ?

image

```patch
Index: core/src/mindustry/ui/dialogs/JoinDialog.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/core/src/mindustry/ui/dialogs/JoinDialog.java b/core/src/mindustry/ui/dialogs/JoinDialog.java
--- a/core/src/mindustry/ui/dialogs/JoinDialog.java	(revision 43c7832f0a6b4df6be5408a0ab075097e2ae8eaf)
+++ b/core/src/mindustry/ui/dialogs/JoinDialog.java	(date 1687227381701)
@@ -418,7 +418,7 @@
             Table[] groupTable = {null, null};
 
             if(group.prioritized){
-                addHeader(groupTable, group, hidden);
+                addHeader(groupTable, group, hidden, false);
             }
             //table containing all groups
             for(String address : group.addresses){
@@ -434,9 +434,7 @@
                         || (res.modeName != null && res.modeName.toLowerCase().contains(serverSearch)))) return;
 
                     //add header
-                    if(groupTable[0] == null){
-                        addHeader(groupTable, group, hidden);
-                    }
+                    addHeader(groupTable, group, hidden, true);
 
                     addCommunityHost(res, groupTable[1]);
 
@@ -447,9 +445,15 @@
         }
     }
 
-    void addHeader(Table[] groupTable, ServerGroup group, boolean hidden){ // outlined separately
+    void addHeader(Table[] groupTable, ServerGroup group, boolean hidden, boolean visible){ // outlined separately
+        if(groupTable[0] != null) {
+            groupTable[0].visible = true;
+            return;
+        }
+
         global.table(t -> groupTable[0] = t).fillX().left().row();
 
+        groupTable[0].visible = visible;
         groupTable[0].table(head -> {
             Color col = group.prioritized ? Pal.accent : Color.lightGray;
             if(!group.name.isEmpty()){
```

Comment (Anuken) [reply core/src/mindustry/ui/dialogs/JoinDialog.java:421]: The UI system is pretty awful at dynamically sizing things based on visibility, so the best option here is probably to use a Collapser without an animation.

**Comment by Phinner** [comment]
@BasedUser, I made PR (https://github.com/BasedUser/Mindustry/pull/7) fixing https://github.com/Anuken/Mindustry/pull/8720#discussion_r1233370956.

**Comment by BasedUser** [comment]
re: [f717108](https://github.com/Anuken/Mindustry/pull/8720/commits/f7171088020e5350c3338e8cf13f73ad4a0ba5e4)
this used to happen

https://github.com/Anuken/Mindustry/assets/21104932/818a3a0d-fd0a-4c24-83e2-ae95729eb03b

**Comment by BasedUser** [comment]
tested with modern server list, looks fine

**Pull request merged**